src/periphery/CauldronFeeWithdrawer.sol

```diff
@@ -0,0 +1,284 @@
+        address[] memory cauldrons,
+        uint8[] memory versions,
+        bool[] memory enabled
+    ) external onlyOwner {
+        for (uint256 i = 0; i < cauldrons.length; i++) {
+            _setCauldron(cauldrons[i], versions[i], enabled[i]);
+        }
+    }
+
+    function _setCauldron(
+        address cauldron,
+        uint8 version,
+        bool enabled
+    ) private {
+        bool previousEnabled;
+
+        for (uint256 i = 0; i < cauldronInfos.length; i++) {
+            if (cauldronInfos[i].cauldron == cauldron) {
+                cauldronInfos[i] = cauldronInfos[cauldronInfos.length - 1];
+                cauldronInfos.pop();
+                break;
+            }
+        }
+
+        if (enabled) {
+            cauldronInfos.push(
+                CauldronInfo({
+                    cauldron: cauldron,
+                    masterContract: address(ICauldronV1(cauldron).masterContract()),
+                    bentoBox: IBentoBoxV1(ICauldronV1(cauldron).bentoBox()),
+                    version: version
+                })
+            );
+        }
+
+        emit LogCauldronChanged(cauldron, previousEnabled, enabled);
+    }
+
+    function setSwappingRecipient(address recipient, bool enabled) external onlyOwner {
+        emit LogSwappingRecipientChanged(recipient, swappingRecipients[recipient], enabled);
+        swappingRecipients[recipient] = enabled;
+    }
+
+    function setSwapTokenOut(IERC20 token, bool enabled) external onlyOwner {
+        emit LogAllowedSwapTokenOutChanged(token, swapTokenOutEnabled[token], enabled);
+        swapTokenOutEnabled[token] = enabled;
+    }
+
+    function setBridgeableToken(IERC20 token, bool enabled) external onlyOwner {
+        emit LogBridgeableTokenChanged(token, bridgeableTokens[token], enabled);
+        bridgeableTokens[token] = enabled;
+    }
+
+    function setParameters(
+        address _swapper,
+        address _mimProvider,
+        ICauldronFeeBridger _bridger
+    ) external onlyOwner {
+        if (_swapper != swapper) {
+            if (swapper != address(0)) {
+                mim.approve(swapper, 0);
+            }
+
+            mim.approve(_swapper, type(uint256).max);
+            swapper = _swapper;
+        }
+
+        mimProvider = _mimProvider;
+        bridger = _bridger;
+
+        emit LogParametersChanged(_swapper, _mimProvider, _bridger);
+    }
+
+    function setBentoBox(IBentoBoxV1 bentoBox, bool enabled) external onlyOwner {
+        bool previousEnabled;
+
+        for (uint256 i = 0; i < bentoBoxes.length; i++) {
+            if (bentoBoxes[i] == bentoBox) {
+                bentoBoxes[i] = bentoBoxes[bentoBoxes.length - 1];
+                bentoBoxes.pop();
+                previousEnabled = true;
+                break;
+            }
+        }
+
+        if (enabled) {
+            bentoBoxes.push(bentoBox);
+        }
+
+        emit LogBentoBoxChanged(bentoBox, previousEnabled, enabled);
+    }
+
+    function rescueTokens(
```

Comment:
maybe we could make this onlyOperator, so we can still manage the cauldrons while the MSpellSender can access this function